openapi3/schema.go

```diff
@@ -1231,3 +1273,113 @@ func isSliceOfUniqueItems(xs []interface{}) bool {
 func unsupportedFormat(format string) error {
 	return fmt.Errorf("Unsupported 'format' value '%s'", format)
 }
+
+func invalidDefaultValue(s *Schema, err error) error {
+	return fmt.Errorf("default value error when validating schema %s (default: %v, error: %s)", "*name*", s.Default, err.Error())
+}
+
+func validateDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	if err := f(s); err != nil {
+		return invalidDefaultValue(s, err)
+	}
+	return nil
+}
+
+func validateStringDefaultProperty(s *Schema, f func(s *Schema) error) error {
+	if s.Default == nil {
+		return nil
+	}
+	_, ok := s.Default.(string)
+	if !ok {
+		return invalidDefaultValue(s, errors.New("not a string"))
```

Comment:
Would it be terrible to accept things that are "stringable"?  (eg do a few number checks, etc)

Reply:
I was trying to keep the intention of the field declaration. If I specify that a field is string, would we rather indicate an error if someone specified 3.14159 instead of "3.14159" in the json/yaml?
I opted to do so because I'd rather the user know about this when they craft the spec than silently convert at runtime, but am happy to consider the alternatives

Reply:
I ask because in general JSON isn't terribly strongly typed and it is a bit of an unexpected behavior to see 1 not being interpreted as "1" automatically.  I suspect other folks who are used to untyped languages will fell the same.

Reply:
would you still like this to be the case even though this is used specifically for defining a spec?
Again, I am happy to change.

Reply:
Correct me if I'm wrong but doesn't openapi3 not specify struct typing of the defaults?  If it does then leave this.  If it doesn't we should go to a best effort to convert things.

Reply:
json-schema is very strict this way too, `type: string` with a value of 42 won't validate. coming from json-schema i'd be surprised about conversions.

Reply:
left it as is